### DISCUSSION TITLE
Tor relays now up to 4 instances per IP

### DISCUSSION THREAD

**Discussion by bolle732** [Enkidu-6/tor-ddos#77] [General]
Seems one has to upper the connection limits from 2 to 4 as Tor relays are now officially allowed to run 4 instances per IP...

**Comment by bolle732** [comment]
https://gitlab.torproject.org/tpo/core/tor/-/issues/40744

**Comment by Enkidu-6** [comment]
Hi, 

Not planning to make any changes any time soon as it's not justified yet for the following reasons:

- There are currently 13 Relay operators who are taking advantage of this
- All of them combined have added  84 extra relays representing 44 IP addresses out of 6341 relays. That's about 1%
- Majority of added relays are - by a large margin -Exit nodes with zero Guard probability and zero Middle relay probability which means  we're the ones building circuits with them as they are the last hop. I may be wrong about this and it may still be necessary for them to build circuits with us but I'll figure that out. 
- All those IP addresses already show up in our dual-or relay ipset which means they are not banned and will still get their two connections.
- For the third or fourth connection, if ever needed, they can and will simply move on to another relay without affecting their operation or yours and will create no problems for the health of the Tor network, their relays or yours.
- Our rules will not affect outgoing connections, which means you can build 2, 3 or 4 circuits with them so we're not starving them either.
- Allowing 4 connections for such a minuscule number of relays will take too much unnecessary resources on everyone's servers with no noticeable or justifiable benefit to the Tor network as a whole.

I'll be watching those numbers and if they grow to a reasonable count, I'll gladly make modifications to the rules.

**Comment by Enkidu-6** [comment]
Hello again @bolle732 

Just an update so you know I haven't forgotten about your concern. Please follow the discussion I started on the mailing list to stay current on the subject.

[tor-relays- Questions about 4 Relays per IP and the ddos mitigation scripts](https://forum.torproject.net/t/tor-relays-questions-about-4-relays-per-ip-and-the-ddos-mitigation-scripts/6463)

**Comment by bolle732** [comment]
Hi @Enkidu-6 

Thanks a lot for the link. I'll keep observing. It's nice to read your opinion and see how this evolves. Learning for all around ;-)

**Comment by Enkidu-6** [comment]
@bolle732 Just to give you another update. Version 6.00 is out and it now deals with your concerns:
https://github.com/Enkidu-6/tor-ddos/releases

**Comment by bolle732** [comment]
@Enkidu-6 Great to hear from you with good news! I'll check and adapt your changes into my script too. Thanks a lot :-)
